### ISSUE TITLE
suggestion: Shader snapshots (history)

### ISSUE THREAD

**Issue by patuwwy** [patuwwy/ShaderToy-Chrome-Plugin#25]
Save shader current state (code/inputs) on local machine. 

It will allow to keep shader history and (view/restore) shader if necessary.

**Comment by ttoinou** [comment]
An automatic way to connect your shader to a .git would be great

**Comment by patuwwy** [comment]
You mean private github repository? Sounds nice but requires github account

**Comment by EthanLowenthal** [comment]
I think ttoinou is suggesting making a local git repository with the shader history. Hard to do in an extention, but possible... There is a pure js implementation of git but it runs on node...

**Comment by ttoinou** [comment]
Yes ! Would be practical. Or just automatic backups create a history and we can make diffs on our own :)

**Comment by patuwwy** [comment]
My first idea was to save to the database in the browser, e.g. every 5 minutes, or every successful compilation. With option do save/restore backup to/from file
